pkg/api/types.go

```diff
@@ -2121,6 +2121,16 @@ type PodSpec struct {
 	// file if specified. This is only valid for non-hostNetwork pods.
 	// +optional
 	HostAliases []HostAlias
+	// If specified, indicates the pod's priority. "system" is a special keyword
```

Comment:
We actually wanted to use "system" for critical system pods. I am fine with the idea of the making this a special keyword that pods cannot use, but then we will need to ship Kubernetes with at least one more predefined priority for system pods.
The PR to add PriorityClass is branched off of this branch and currently lives in my repo until this one is merged: https://github.com/bsalamat/kubernetes#1